### ISSUE TITLE
Video MiniGPT4

### ISSUE THREAD

**Issue by pixeli99** [OpenGVLab/Ask-Anything#16]
For minigpt4, can it be realized directly using video embedding?
Just like,

```python
query_tokens = self.query_tokens.expand(image_embeds.shape[0], -1, -1)
            query_output = self.Qformer.bert(
                query_embeds=query_tokens,
                encoder_hidden_states=image_embeds,
                encoder_attention_mask=image_atts,
                return_dict=True,
)
# [bs, num_frames, 32, 768] -> [bs, num_frames, 32, 768] -> [bs, num_frames * 32, 768]
video_out = self.perceive(query_output.last_hidden_state.view(b, t, query_tokens.shape[-2], query_tokens.shape[-1])).flatten(1, 2)
inputs_llama = self.llama_proj(video_out)
```
As for the `self.perceive`, Maybe a simple self attention will do？
Just like flamingo
```python
class PerceiverResampler(nn.Module):
    def __init__(
        self,
        *,
        dim,
        depth,
        dim_head = 64,
        heads = 8,
        num_latents = 64,
        num_media_embeds = 4,
        ff_mult = 4
    ):
        super().__init__()
        self.latents = nn.Parameter(torch.randn(num_latents, dim))
        self.media_pos_emb = nn.Parameter(torch.randn(num_media_embeds, 1, dim))

        self.layers = nn.ModuleList([])
        for _ in range(depth):
            self.layers.append(nn.ModuleList([
                PerceiverAttention(dim = dim, dim_head = dim_head, heads = heads),
                FeedForward(dim = dim, mult = ff_mult)
            ]))

        self.norm = nn.LayerNorm(dim)

    def forward(self, x):
        if x.ndim == 3:
            x = rearrange(x, 'b n d -> b 1 n d')

        times = x.shape[1]
        x = x + self.media_pos_emb[:times]

        latents = repeat(self.latents, 'n d -> b m n d', b = x.shape[0], m = x.shape[1])

        for attn, ff in self.layers:
            latents = attn(x, latents) + latents
            latents = ff(latents) + latents

        return self.norm(latents)
```

I don't have enough GPUs to verify this idea. Maybe it is very naive. I just put it here and hope to inspire some interested friends.

**Comment by Andy1621** [comment]
Yes, it's a good question!
Actually, we have tried different methods, like randomly selecting one frame, averaging the frames, and merging frames via QFormer. Those methods work but are not so sensitive to temporal information without video fine-tuning. We just introduce a simple temporal template since it seems more sensitive to temporal information.

**Comment by pixeli99** [comment]
Yes, it seems that we need to train on video to achieve better results. In short, if LLM can understand the behavior of a time series, I think it would be very interesting. 
Currently, `Ask-Anything` is not as good as it is (or there may be issues with my usage), such as turning a car, and so on.

**Comment by Andy1621** [comment]
Yes! We are trying to train a video model for better temporal reasoning!

**Comment by Xinxinatg** [comment]
> Yes! We are trying to train a video model for better temporal reasoning!

any timeline proposed yet for the new module? rumor says that the gpt5 might incorporate video understanding, I think your work can be easily extended for better performance like what was suggested in this thread? or maybe using [videomae](https://github.com/OpenGVLab/VideoMAEv2) as the video encoder? from my understanding, the temporal information would require extra training which involves the allignment of spacial information accross different frames, that's why the static information obtained from static images won't be too helpful for video understanding

**Comment by YiyangZhou** [comment]
Thank you very much for your work, it's been a really novel and interesting experience! As for this question, I just happen to have thought of our newly proposed model mPLUG-Owl. I have tested it for a long time and found that its visual ability is quite strong. Meanwhile, our visual system is designed to support interleaved text and image as well as multiple images. Would it be possible to implement this idea on our model? I'm really looking forward to hearing from your team! 😊
mPLUG-Owl：https://github.com/X-PLUG/mPLUG-Owl

**Comment by Andy1621** [comment]
@YiyangZhou @hangzhang-nlp  Thanks for your great jobs! These days we are busy with other things. 
I have seen that mPLUG-owl supports video now, but it seems to simply concat the image embedding. And Video-LLaMA only uses WebVid in the first stage. 
Now we have released the [video data for instruction tuning](https://github.com/OpenGVLab/Ask-Anything/tree/main/video_chat). Maybe your teams can use it for better results! If you find them work or do not work, don't hesitate to tell me 😁!

**Comment by lixin4ever** [comment]
@Andy1621 Just go through your technical report and project page, it's definitely a great job in terms of technical contributions and resource contributions👍👍👍 We really appreciate your kindness in releasing the video instruction-tuning data so promptly (and also letting us know :)   Will try your instruction tuning data soon and for sure mention your work in our repo.
